iothub/device/samples/how to guides/DeviceReconnectionSample/CustomRetryPolicy.cs

```diff
@@ -0,0 +1,74 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using Microsoft.Azure.Devices.Client.Exceptions;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Azure.Devices.Client.Samples
+{
+    /// <summary>
+    /// An exponential backoff with jitter based retry policy that retries on additional exceptions
+    /// not covered by the SDK, because in this application we wish to run as long as possible.
+    /// </summary>
+    internal class CustomRetryPolicy : IRetryPolicy
+    {
+        private const int MaxRetryCount = int.MaxValue;
+        private const int MaxExponent = 22; // Avoid integer overlow (max of 30) and clamp max wait to just over 1 hour (2^22 = 1.16 hours).
+
+        private readonly Random _rng = new();
```

Comment:
This may be a bit too complex of a custom policy for 101 level samples. Is it just because you want this to be a good policy in case a user copy/pastes this?

Reply:
Pretty much. I could simply do a constant retry interval, though. Thoughts?

Reply:
We do use this sample with customers to tell them how to write their app. It is definitely more than a 100-level sample. That is why it is not in the getting started folder.